zerver/actions/streams.py

```diff
@@ -69,6 +70,11 @@
 def do_deactivate_stream(
     stream: Stream, log: bool = True, *, acting_user: Optional[UserProfile]
 ) -> None:
+    # Error if this stream is used for reporting messages. Note that this also works for
+    # merge_streams admin command on the command-line.
+    if Realm.objects.filter(id=stream.realm_id, reportmsg_stream=stream).exists():
```

Comment:
`if stream.realm.reportmsg_stream_id == stream.id` is probably more efficient since the realm will be cached.

Reply:
DONE - was playing it safe... thx for the tip.